cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/OAuthAccessTokenResponseGenerator.java

```diff
@@ -0,0 +1,118 @@
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This is {@link OAuthAccessTokenResponseGenerator}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@RefreshScope
+@Component("oauthAccessTokenResponseGenerator")
+public class OAuthAccessTokenResponseGenerator implements AccessTokenResponseGenerator {
+    protected transient Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * The JSON factory.
+     */
+    protected final JsonFactory jsonFactory = new JsonFactory(new ObjectMapper());
+
+    /**
+     * The Resource loader.
+     */
+    @Autowired
+    protected ResourceLoader resourceLoader;
+
+    @Override
+    public void generate(final HttpServletRequest request,
+                         final HttpServletResponse response,
+                         final OAuthRegisteredService registeredService,
+                         final Service service,
+                         final AccessToken accessTokenId,
+                         final RefreshToken refreshTokenId,
+                         final long timeout) {
+
+        if (registeredService.isJsonFormat()) {
+            response.setContentType(MediaType.APPLICATION_JSON_VALUE);
+            try (final JsonGenerator jsonGenerator = this.jsonFactory.createGenerator(response.getWriter())) {
+                jsonGenerator.writeStartObject();
+                generateJsonInternal(request, response, jsonGenerator, accessTokenId,
+                        refreshTokenId, timeout, service, registeredService);
+                jsonGenerator.writeEndObject();
+            } catch (final Exception e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            generateTextInternal(request, response, accessTokenId, refreshTokenId, timeout);
+        }
+    }
+
+    /**
+     * Generate text internal.
+     *
+     * @param request        the request
+     * @param response       the response
+     * @param accessTokenId  the access token id
+     * @param refreshTokenId the refresh token id
+     * @param timeout        the timeout
+     */
+    protected void generateTextInternal(final HttpServletRequest request,
+                                        final HttpServletResponse response,
+                                        final AccessToken accessTokenId,
+                                        final RefreshToken refreshTokenId,
+                                        final long timeout) {
+        String text = String.format("%s=%s&%s=%s", OAuthConstants.ACCESS_TOKEN, accessTokenId.getId(),
+                OAuthConstants.EXPIRES, timeout);
+        if (refreshTokenId != null) {
+            text += '&' + OAuthConstants.REFRESH_TOKEN + '=' + refreshTokenId.getId();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuffer over += for concatenating strings](https://www.codacy.com/app/mmoayyed/cas/file/2516371513/issues/source?bid=3256115&fileBranchId=3273836#l86)